datafusion/core/src/physical_plan/hash_utils.rs

```diff
@@ -0,0 +1,830 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
I don't understand this file -- it looks like maybe it was previously moved into physical-expr and maybe a merge conflict caused it to end up here?

```
find . -name 'hash_utils.rs'
./datafusion/core/src/physical_plan/hash_utils.rs
./datafusion/physical-expr/src/hash_utils.rs
``

Reply:
Hello @alamb! I have been looking into the merges and I think that is exactly what happened. Indeed, this file was present in version 13, but it is no longer there. I had to change `/datafusion/core/src/physical_plan/joins/hash_join.rs` and I ended up modifying `datafusion/core/src/physical_plan/hash_utils.rs` as well. Then I wasn't sure if this file had been eliminated or not, and while solving the merge conflicts I thought it had not been, but I was wrong. 

It seems clear to me now that this file has to be removed and everything will be solved. But maybe I am missing something? I will push the new version without this file, please let me know what you think.

And thank you very much for your words and for going through my changes! We are happy to cooperate with you!